changelogs/fragments/reserved_vars_fix.yml

```
@@ -0,0 +1,3 @@
+bugfixes:
+  - Ansible now warns for all reserved vars being defined, they were currently being silently overwritten, but now will give an adequate warning.
+  - Ansible no longer allows clobering of some internal variables (group_names, inventory_hostname_short, inventory_hostname, ansible_local, ansible_facts, inventory_file, inventory_dir)
```

I could see this change breaking content, mostly ansible_local/ansible_facts. I don't think this aspect is as safe to backport as the rest.

```suggestion
  - Ansible no longer allows clobbering of some internal variables (group_names, inventory_hostname_short, inventory_hostname, ansible_local, ansible_facts, inventory_file, inventory_dir)
```